src/code/z_scene.c

```diff
@@ -378,30 +378,30 @@ void Scene_HeaderCmdSkyboxDisables(PlayState* play, SceneCmd* cmd) {
 
 // SceneTableEntry Header Command 0x10: Time Settings
 void Scene_HeaderCmdTimeSettings(PlayState* play, SceneCmd* cmd) {
-    if (cmd->timeSettings.hour != 0xFF && cmd->timeSettings.min != 0xFF) {
+    if ((cmd->timeSettings.hour != 0xFF) && (cmd->timeSettings.min != 0xFF)) {
         gSaveContext.skyboxTime = gSaveContext.save.time =
-            (u16)(((cmd->timeSettings.hour + (cmd->timeSettings.min / 60.0f)) * 60.0f) / 0.021972656f);
+            CLOCK_TIME_ALT2_F(cmd->timeSettings.hour, cmd->timeSettings.min);
     }
 
-    if (cmd->timeSettings.unk6 != 0xFF) {
-        play->envCtx.timeIncrement = cmd->timeSettings.unk6;
+    if (cmd->timeSettings.timeSpeed != 0xFF) {
+        play->envCtx.sceneTimeSpeed = cmd->timeSettings.timeSpeed;
     } else {
-        play->envCtx.timeIncrement = 0;
+        play->envCtx.sceneTimeSpeed = 0;
     }
 
-    if ((gSaveContext.save.inventory.items[SLOT_OCARINA] == ITEM_NONE) && (play->envCtx.timeIncrement != 0)) {
-        play->envCtx.timeIncrement = 5;
+    if ((gSaveContext.save.inventory.items[SLOT_OCARINA] == ITEM_NONE) && (play->envCtx.sceneTimeSpeed != 0)) {
+        play->envCtx.sceneTimeSpeed = 5;
     }
```

Comment:
```suggestion
    // Increase time speed during zeroth cycle
    if ((gSaveContext.save.inventory.items[SLOT_OCARINA] == ITEM_NONE) && (play->envCtx.sceneTimeSpeed != 0)) {
        play->envCtx.sceneTimeSpeed = 5;
    }
```

Reply:
Isn't this commonly referred to as first cycle? And there's nothing explicitly tying it to cycles, time being faster is only dependent on not having the ocarina.

Reply:
Yes, but I feel like zeroth is a better name myself. It is true that the game is inconsistent in how it checks for this (actually having Ocarina vs having anything in the Ocarina slot), but since getting Ocarina back is the end of zeroth cycle, it makes sense. Unless you want zeroth cycle to refer to pre-Clock Town, which might be a better idea.

Reply:
I still feel like getting ocarina back at the end of the zeroth/first cycle is still higher-order game logic, as there's no actual code here to check which cycle you're on. For example, if something happened to remove the ocarina, then the scene speed would increase again, despite it not being zeroth/first cycle

Reply:
Yeah, the state "first cycle" usually refers to not having Ocarina in Ocarina slot, somewhat counterintuitively.

Reply:
What other game logic directly changes on not having an ocarina item/empty ocarina slot? I'll look into it, that might change my perspective depending

Reply:
There's special logic in Play for which version of Termina Field to load, I think is the usual example.

Reply:
```c
s32 ObjTokeidai_IsPostFirstCycleFinalHours(ObjTokeidai* this, PlayState* play) {
    if (gSaveContext.save.inventory.items[SLOT_OCARINA] == ITEM_NONE) {
        return false;
    }
```

```c
/**
 * Gets the relevant text ID for Tatl hints in first cycle.
 */
u16 QuestHint_GetTatlTextId(PlayState* play) {
    if (INV_CONTENT(ITEM_OCARINA) == ITEM_OCARINA) {
        return 0;
    }
```
Yeah, it does show up in a few places. Also, the repo does seem to be using "first cycle" instead of "zeroth" cycle currently (it can change, but it's just an observation)

Reply:
Yeah, fair enough to make it consistent for now, we can get back to it once we have all the code.